tests/test_closing.py

```diff
@@ -2392,35 +2392,45 @@ def try_pay():
 
-    # Graph of coin_move events we expect
-    expected_1 = {
-        '0': [('wallet', ['deposit'], ['withdrawal'], 'A')],
-        # This is ugly, but this wallet deposit is either unspent or used
-        # in the next channel open
-        'A': [('wallet', ['deposit'], None, None), ('cid1', ['channel_open', 'opener'], ['channel_close'], 'B')],
-        'B': [('wallet', ['deposit'], None, None), ('cid1', ['htlc_timeout'], ['to_wallet'], 'C')],
-        'C': [('wallet', ['deposit'], None, None)],
-    }
+    # Graph of coin_move events we expect!
+    if anchors:
+        expected_1 = {
+            # Initial wallet deposit
+            '0': [('wallet', ['deposit'], ['withdrawal'], 'A')],
+            # Funding tx
+            'A': [('wallet', ['deposit'], None, None), ('cid1', ['channel_open', 'opener'], ['channel_close'], 'B')],
+            # Commitment tx
+            'B': [('wallet', ['deposit'], None, None), ('cid1', ['htlc_timeout'], ['to_wallet'], 'C'), ('external', ['anchor'], None, None), ('wallet', ['anchor', 'ignored'], None, None)],
+            # HTLC timeout tx
+            'C': [('wallet', ['deposit'], None, None)],
+        }
 
-    expected_2 = {
-        'A': [('cid1', ['channel_open'], ['channel_close'], 'B')],
-        'B': [('external', ['to_them'], None, None), ('external', ['htlc_timeout'], None, None)],
-    }
+        expected_2 = {
+            # Funding tx
+            'A': [('cid1', ['channel_open'], ['channel_close'], 'B')],
+            # Commitment tx
+            'B': [('external', ['to_them'], None, None), ('external', ['htlc_timeout'], None, None), ('external', ['anchor'], None, None), ('wallet', ['anchor', 'ignored'], None, None)],
+        }
+    else:
+        expected_1 = {
+            '0': [('wallet', ['deposit'], ['withdrawal'], 'A')],
+            # This is ugly, but this wallet deposit is either unspent or used
+            # in the next channel open
+            'A': [('wallet', ['deposit'], None, None), ('cid1', ['channel_open', 'opener'], ['channel_close'], 'B')],
+            'B': [('wallet', ['deposit'], None, None), ('cid1', ['htlc_timeout'], ['to_wallet'], 'C')],
+            'C': [('wallet', ['deposit'], None, None)],
+        }
 
-    if anchors:
-        expected_1['B'].append(('external', ['anchor'], None, None))
-        expected_2['B'].append(('external', ['anchor'], None, None))
-        expected_1['B'].append(('wallet', ['anchor', 'ignored'], None, None))
-        expected_2['B'].append(('wallet', ['anchor', 'ignored'], None, None))
+        expected_2 = {
+            'A': [('cid1', ['channel_open'], ['channel_close'], 'B')],
+            'B': [('external', ['to_them'], None, None), ('external', ['htlc_timeout'], None, None)],
+        }
 
-    # FIXME: Why does this fail?
-    if not anchors:
-        tags = check_utxos_channel(l1, [channel_id], expected_1)
-        check_utxos_channel(l2, [channel_id], expected_2, tags)
+    tags = check_utxos_channel(l1, [channel_id], expected_1)
```

Comment:
we can now re-activate this part of the test, since we're no longer sending out anchorspend txs, which were causing this to fail.